app/src/main/java/org/fossasia/openevent/general/event/EventService.kt

```diff
@@ -63,20 +63,12 @@ class EventService(private val eventApi: EventApi, private val eventDao: EventDa
         }
     }
 
-    fun getSimilarEvents(id: Long): Flowable<List<Event>> {
-        val eventsFlowable = eventDao.getAllSimilarEvents(id)
-        return eventsFlowable.switchMap {
-            if (it.isNotEmpty())
-                eventsFlowable
-            else
-                eventTopicApi.getEventsUnderTopicId(id)
-                        .toFlowable()
-                        .map {
-                            eventDao.insertEvents(it)
-                        }
-                        .flatMap {
-                            eventsFlowable
-                        }
+    fun getSimilarEvents(id: Long, savedLocation: String): Single<List<Event>> {
+        return eventTopicApi.getEventsUnderTopicId(id, "name", savedLocation).flatMap { apiList ->
+            val eventIds = apiList.map { it.id }.toList()
+            eventDao.getFavoriteEventWithinIds(eventIds).flatMap { favIds ->
+                updateFavorites(apiList, favIds)
```

Comment:
don't blindly copy code. Take your time before pushing it

Reply:
Actually the same process is being done and that's why it looks like it is being blindly copied :smile: . We are loading events based on event topic and updating favorites by mapping. Isn't mapping required?

Reply:
why are you returning favorite events from db

Reply:
you have to display similar events now only those events which are favorite in similar events will be displayed